tests/NexusMods.DataModel.SchemaVersions.Tests/SchemaFailsafes.cs

```diff
@@ -0,0 +1,86 @@
+using System.Text;
+using FluentAssertions;
+using NexusMods.Hashing.xxHash3;
+using NexusMods.MnemonicDB.Abstractions;
+
+namespace NexusMods.DataModel.Migrations.Tests;
+
+public class SchemaFailsafes(IConnection connection)
+{
+
+    [Fact]
+    public async Task SchemaFingerprintHasntChanged()
+    {
+        var db = connection.Db;
+        var records = db.AttributeCache.AllAttributeIds
+            .OrderBy(id => id.Id)
+            .Select(id =>
+                {
+                    var aid = db.AttributeCache.GetAttributeId(id);
+                    return new string[]
+                    {
+                        id.ToString(),
+                        db.AttributeCache.GetValueTag(aid).ToString(),
+                        db.AttributeCache.IsIndexed(aid).ToString(),
+                        db.AttributeCache.IsCardinalityMany(aid).ToString(),
+                        db.AttributeCache.IsNoHistory(aid).ToString(),
```

Comment:
Same as before, I think this should include whether the attribute is optional or not.

Reply:
I can add this in, in the future. Right now this is pulling data from the MnemonicDB internal schema (in the datomstore itself), I'm fine with adding optional as another field on there, but I need to do another release of MnemonicDB first.

Reply:
That's fine for this PR, but it's something we'd want sooner rather than later.